### PR TITLE
Better yr

### PR DESCRIPTION
**Pull request by henriklovhaug**
Needs a few fixes before it is ready for production
* Match sunny/cloudy/rainy to icons
* Pass evaluation of design choices

### CODE REVIEW

File: ababot/bot/Cargo.toml [hunk 1]
```diff
@@ -55,4 +55,8 @@ rayon = "1.6"
 # GPU
 gpgpu = { git = "https://github.com/UpsettingBoy/gpgpu-rs", branch = "dev" }
 bytemuck = "1.12.3"
-bichannel = "0.0.4"
\ No newline at end of file
+bichannel = "0.0.4"
```
Comment (ghost) [inline ababot/bot/Cargo.toml]: Do we still use bichannel?

**Review by mack-stefan — DISMISSED** [review]

File: ababot/bot/src/background_tasks/yr/image.rs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            }
+        }
+    }
+
+    flip_vertical_in_place(&mut imgbuf);
+
+    let renderer = TextRenderer::default();
+    let unit_image = renderer
+        .render_text_to_png_data(format!("{}mm", scalar), 32, Color::new(255, 255, 255))
+        .unwrap();
```
Comment (ghost) [inline ababot/bot/src/background_tasks/yr/image.rs:124]: O.o

File: ababot/bot/src/background_tasks/yr/image.rs [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    let unit_image_width = unit_image.width() as i64;
+    image::imageops::overlay(&mut imgbuf, &unit_image, 1000 - unit_image_width, 500);
+
+    let unit_image_2 = renderer
+        .render_text_to_png_data(
+            format!("{}mm", scalar / 2f32),
+            32,
+            Color::new(255, 255, 255),
+        )
+        .unwrap();
```
Comment (ghost) [inline ababot/bot/src/background_tasks/yr/image.rs:135]: O.o

File: ababot/bot/src/background_tasks/yr/image.rs [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    let mut imgbuf = RgbaImage::new(IMGX, IMGY);
+
+    let renderer = TextRenderer::default();
+    for i in 0i64..10 {
+        let time = times[i as usize];
+        let time_image = renderer
+            .render_text_to_png_data(time.format("%H").to_string(), 32, Color::new(255, 255, 255))
+            .unwrap();
```
Comment (ghost) [inline ababot/bot/src/background_tasks/yr/image.rs:155]: O.o

File: ababot/bot/src/background_tasks/yr/image.rs [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        let time_image = image::load_from_memory(&time_image.data).unwrap();
+        image::imageops::overlay(&mut imgbuf, &time_image, 100 * i + 40, 400);
+        let degree = degrees[i as usize];
+        let degree_image = renderer
+            .render_text_to_png_data(format!("{}°", degree), 32, Color::new(255, 255, 255))
+            .unwrap();
```
Comment (ghost) [inline ababot/bot/src/background_tasks/yr/image.rs:161]: O.o

File: ababot/bot/src/background_tasks/yr/yr_task.rs [hunk 6]
```diff
@@ -109,23 +90,17 @@ fn parse_weather(response: String) -> Result<Vec<Series>, String> {
...
-    if let Some(s) = serie_at_12 {
-        Ok(s.clone())
-    } else {
-        Err("No weather at 12".into())
-    }
+    Ok(weather.to_vec())
 }
 
+// TODO: Incorporate this to the image
+#[allow(dead_code)]
```
Comment (ghost) [inline ababot/bot/src/background_tasks/yr/yr_task.rs:103]: O.o

**Review by mack-stefan — APPROVED** [review]

**Pull request merged**